desci-contracts/contracts/DpidAliasRegistry.sol

```diff
@@ -0,0 +1,131 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract DpidAliasRegistry is OwnableUpgradeable {
+    uint256 public _firstDpid;
+    uint256 public nextDpid;
+
+    // dpid => codex streamID
+    mapping(uint256 => string) public registry;
+
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function __DpidAliasRegistry_init(uint256 firstDpid) public initializer {
+        OwnableUpgradeable.__Ownable_init();
+        firstDpid = firstDpid;
+        nextDpid = firstDpid;
+    }
+
+    /**
+     * Lookup the codex stream ID of a given dPID
+     * @param dpid the alias to lookup
+     */
+    function lookup(uint256 dpid) public view returns(string memory) {
+        return registry[dpid];
+    }
+
+    event DpidMinted (
+        uint256 dpid,
+        string streamID
+    );
+
+    /**
+     * Claim the next free dPID by pointing it to a codex stream ID
+     * 
+     * @param streamId the codex stream ID to alias
+     */
+    function mintDpid(string calldata streamId) public returns(uint256) { 
```

Comment:
I think there should be a check here to check if the nextDpid is an unmigrated legacy dpid and skips to the next until it finds a free dpid.
Am I missing something? what are your thoughts?

Reply:
@shadrach-tayo This contract starts minting at whatever `firstDpid` we pass to the initializer, so `nextDpid` will always be available. The idea is to simply skip all dPID's that are already minted at initialization, insetad of having to do an extra check on every mint.

Reply:
oh I see